pkg/customresourcestate/registry_factory.go

```diff
@@ -257,22 +244,54 @@ func (c *compiledGauge) Values(v interface{}) (result []eachValue, errs []error)
 
 type compiledInfo struct {
 	compiledCommon
+	labelFromKey string
 }
 
 func (c *compiledInfo) Values(v interface{}) (result []eachValue, errs []error) {
-	if vs, isArray := v.([]interface{}); isArray {
-		for _, obj := range vs {
+	onError := func(err ...error) {
+		errs = append(errs, fmt.Errorf("%s: %v", c.Path(), err))
+	}
+
+	switch iter := v.(type) {
+	case []interface{}:
+		for _, obj := range iter {
 			ev, err := c.values(obj)
 			if len(err) > 0 {
-				errs = append(errs, err...)
+				onError(err...)
 				continue
 			}
 			result = append(result, ev...)
 		}
-		return
+	case map[string]interface{}:
+		value, err := c.values(v)
+		if err != nil {
+			onError(err...)
+			break
+		}
+		// labelFromKey logic
+		for key, val := range iter {
+			if key != "" && c.labelFromKey != "" {
+				n, err := toFloat64(val, false)
+				if err != nil {
+					onError(err)
+					continue
+				}
+				result = append(result, eachValue{
+					Labels: map[string]string{
+						c.labelFromKey: key,
+					},
```

Comment:
this new label needs to be appended to `value.Labels` otherwise we will be missing the labels related to the path

Reply:
Not sure IIUC, but it seems to be done here: https://github.com/rexagod/kube-state-metrics/blob/1871-1868/pkg/customresourcestate/registry_factory.go#L279-L289.

Reply:
it is done correctly if you don't have `labelFromKey` set, but when it is set, you are manually setting label map of the result to just `c.labelFromKey: key` and thus don't take into account the list of labels that was previously created when calling `c.values(v)`.

Reply:
The thing is that you either only add the eachValue from the labelFromKey logic or the one computed by the `values()` method so both are never present together in the final list of eachValue that you return

Reply:
I added logic to check for the `labelFromKey` value in the set of label keys collected beforehand. PTAL the latest fixup commit.